<launch>
    
    <rosparam command="load" file="$(find aloam_ekf_mapping)/config/params_outdoor.yaml" />

    <!--- Velodyne_packets to velodyne_points -->
    <node pkg="velodyne_pointcloud" type="transform_node" name="transform_node">
      <param name="model" value="32E"/>
      <remap from="/velodyne_packets" to="/hdl32e_left/velodyne_packets"/>
      <param name="calibration" value="$(find velodyne_pointcloud)/params/32db.yaml"/>
      <param name="min_range" value="0.4"/>
      <param name="max_range" value="130.0"/>
      <param name="view_direction" value="3.1415926"/>
      <param name="view_width" value="5.4769"/>
    </node>

    <param name="point_filter_num" type="int" value="4"/>
    <param name="mapping_plane_res" type="double" value="0.5"/>
    <param name="lidar_std" type="double" value="0.01"/>
    <param name="cube_side_length" type="double" value="1000" />

    <node pkg="aloam_ekf_mapping" type="pointProcess" name="pointProcess" output="screen" />

    <node pkg="aloam_ekf_mapping" type="ekfMapping" name="ekfMapping" output="screen" />

    <!--<node pkg="aloam_ekf_mapping" type="ekfMapping" name="ekfMapping" output="screen" launch-prefix="xterm -e gdb -ex run args" />-->

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_ekf_mapping)/rviz_cfg/aloam_velodyne.rviz" />
    </group>

</launch>
